src/NServiceBus.Transport.Msmq/MsmqMessageDispatcher.cs

```diff
@@ -33,15 +33,17 @@ public Task Dispatch(TransportOperations outgoingMessages, TransportTransaction
                 throw new Exception("The MSMQ transport only supports unicast transport operations.");
             }
 
+            var transportOperationTasks = new List<Task>(outgoingMessages.UnicastTransportOperations.Count);
+
             foreach (var unicastTransportOperation in outgoingMessages.UnicastTransportOperations)
             {
-                ExecuteTransportOperation(transaction, unicastTransportOperation);
+                transportOperationTasks.Add(ExecuteTransportOperation(transaction, unicastTransportOperation));
             }
 
-            return TaskEx.CompletedTask;
+            return Task.WhenAll(transportOperationTasks);
```

Comment:
If you add something like [this](https://github.com/Particular/NServiceBus.RabbitMQ/blob/develop/src/NServiceBus.Transport.RabbitMQ/Sending/MessageDispatcher.cs#L37), the you can avoid allocating another `Task` with the `WhenAll` call when there is only one message to send.